lib/chef/knife/bootstrap.rb

```diff
@@ -1081,7 +1109,17 @@ def bootstrap_command(remote_path)
         if connection.windows?
           "cmd.exe /C #{remote_path}"
         else
-          "sh #{remote_path}"
+          cmd = "sh #{remote_path}"
+
+          if config[:su_user]
+            # su - USER is subject to required an interactive console
+            # Otherwise, it will raise: su: must be run from a terminal
+            set_transport_options(pty: true)
```

Comment:
can we do this test earlier and set the transport option as we initialize the transport connection, rather than having to mangle the transport options late?

then there would be no need to change the transport options to be an accessor on the train side

Reply:
The reason here to enabled it later, during platform detection some of system like windows raises platform deletection error ie. https://github.com/chef/chef/issues/8683 by default pty options is disabed so during connection established & platform detection enabled pty only when required. 



So once connection got established, for linux node currently we are dealing with `--su-user` option with pty: true.